layer/executables/tar.py

```diff
@@ -0,0 +1,104 @@
+import pickle  # nosec: import_pickle
+import sys
+import tarfile
+import tempfile
+from pathlib import Path
+from typing import Any, Callable, List, Optional
+
+import cloudpickle  # type: ignore
+
+
+MODEL_TRAIN_ENTRYPOINT_FILE = Path(__file__).parent / "model" / "entrypoint.py"
+EXTRACTOR_PIP_DEPENDENCIES = ["cloudpickle"]
+
+
+def build_executable_tar(
+    path: Path,
+    function: Callable[..., Any],
+    entrypoint: Path,
+    resources: Optional[List[Path]] = None,
+    pip_dependencies: Optional[List[str]] = None,
+) -> None:
+
+    # add pip dependencies needed by the extractor
+    pip_dependencies = pip_dependencies or []
+    pip_dependencies.extend(EXTRACTOR_PIP_DEPENDENCIES)
```

Comment:
To make sure we have reproducible tars between the different layer versions and local and remote builds, I suggest we include the layer package itself which was used to build the tar.

Reply:
It somehow messed up on the CI when I change this and I don't want to waste time anymore on debugging that one, merging this and will address this later on